worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+            self.multiworld.get_location(LocationNames.Q_Goblin_Diplomacy, self.player).can_reach(state)
+        ))
+        add_rule(self.multiworld.get_location(LocationNames.QP_Pirates_Treasure, self.player), lambda state: (
+            self.multiworld.get_location(LocationNames.Q_Pirates_Treasure, self.player).can_reach(state)
+        ))
+        add_rule(self.multiworld.get_location(LocationNames.QP_Rune_Mysteries, self.player), lambda state: (
+            self.multiworld.get_location(LocationNames.Q_Rune_Mysteries, self.player).can_reach(state)
+        ))
+        add_rule(self.multiworld.get_location(LocationNames.QP_Misthalin_Mystery, self.player), lambda state: (
+            self.multiworld.get_location(LocationNames.Q_Misthalin_Mystery, self.player).can_reach(state)
+        ))
+        add_rule(self.multiworld.get_location(LocationNames.QP_Corsair_Curse, self.player), lambda state: (
+            self.multiworld.get_location(LocationNames.Q_Corsair_Curse, self.player).can_reach(state)
+        ))
+        add_rule(self.multiworld.get_location(LocationNames.QP_X_Marks_the_Spot, self.player), lambda state: (
+            self.multiworld.get_location(LocationNames.Q_X_Marks_the_Spot, self.player).can_reach(state)
+        ))
+        add_rule(self.multiworld.get_location(LocationNames.QP_Below_Ice_Mountain, self.player), lambda state: (
+            self.multiworld.get_location(LocationNames.Q_Below_Ice_Mountain, self.player).can_reach(state)
+        ))
+
+        # place "Victory" at "Dragon Slayer" and set collection as win condition
+        self.multiworld.get_location(LocationNames.Q_Dragon_Slayer, self.player) \
+            .place_locked_item(self.create_event("Victory"))
+
+        for location_name, location in self.location_name_to_data.items():
+            location_row = self.location_rows_by_name[location_name]
+            # Set up requirements for region
+            for region_required_name in location_row.regions:
+                region_required = self.region_name_to_data[region_required_name]
+                add_rule(location, lambda state, region_required=region_required: state.can_reach(region_required, None, self.player))
+            for skill_req in location_row.skills:
+                add_rule(location, lambda state, skill_req=skill_req: self.can_reach_skill(state, skill_req.skill, skill_req.level))
+            for item_req in location_row.items:
+                add_rule(location, lambda state, item_req=item_req: state.has(item_req, self.player))
+            if location_row.qp != 0:
+                add_rule(location, lambda state, location_row=location_row: self.quest_points(state) > location_row.qp)
+        self.multiworld.completion_condition[self.player] = lambda state: (state.has("Victory", self.player))
+
+    def create_region(self, name: str) -> "Region":
+        region = Region(name, self.player, self.multiworld)
+        self.region_name_to_data[name] = region
```

Comment:
You don't need to cache regions yourself anymore, `multiworld.get_region` has been greatly sped up and has its own cache now

Disregard if you use this dict somewhere to iterate, but it didn't seem like it to me